.github/workflows/workflow.yml

```diff
@@ -74,14 +74,50 @@ jobs:
     - name: Build documentation website
       run: |
         make --directory ./docs html
-       
-    - name: Publish documentation
-      uses: s0/git-publish-subdir-action@v2.2.0
-      if: contains(github.ref, 'master')
-      env:
-        REPO: self
-        BRANCH: gh-pages
-        FOLDER: docs/build/html
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - uses: actions/upload-artifact@v1
+      with:
+        name: docs
+        path: docs/build/html
+
+  deploy-docs:
+    needs: [tests, docs]
+    runs-on: ubuntu-latest
+    if: contains(github.ref, 'master')
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v1
+        with:
+          name: docs
+          path: docs/build/html
+      - name: Publish documentation
+        uses: s0/git-publish-subdir-action@v2.2.0
+        env:
+          REPO: self
+          BRANCH: gh-pages
+          FOLDER: docs/build/html
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
DId you maybe figure out where this token is coming from and who does it belong to?

Reply:
It is coming from Github Actions